### PR TITLE
[PyTorch] Avoid origin PyTorch methods inference

### PR DESCRIPTION
**Pull request by dkurt**
* We still have to track tensors shapes because of `torch.split` - there is no option to define number of outputs without shapes

### CODE REVIEW

File: modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py [hunk 1]
```diff
@@ -95,78 +103,79 @@ def tensor(self):
...
-        return self._value.dim()
+        return len(self.dynamic_shape)
+    
+    @property
+    def shape(self):
+        return tuple(self.dynamic_shape)
 
     def size(self, dim=None):
-        return self._value.size(dim) if dim else self._value.size()
+        return self.dynamic_shape[dim] if dim else self.shape
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py:116]: Maybe `if dim not None`?

File: modules/mo_pytorch/mo_extensions/load/pytorch/detectron2/retinanet.py [hunk 2]
```diff
@@ -6,11 +6,6 @@
 from mo.utils.error import Error
 
 def inference(model, func, anchors, pred_logits, pred_anchor_deltas, image_sizes):
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/load/pytorch/detectron2/retinanet.py:8]: `func` is unused parameter

Comment (dkurt) [reply modules/mo_pytorch/mo_extensions/load/pytorch/detectron2/retinanet.py:8]: Yes, but it's used in other hooks:

```python
def forward(model, forward, batched_inputs):
    return forward([{'image': batched_inputs}])
```

**Review by likholat — APPROVED** [review]
